src/web/hooks/use-request-modal.ts

```diff
@@ -1,8 +1,19 @@
+import { useOpenAiGlobal } from "./use-openai-global.js";
+
 /**
  * Triggers a modal containing the widget rendered in display mode "modal"
  */
 export function useRequestModal() {
-  return (options: { title: string }) => {
+  const displayMode = useOpenAiGlobal("displayMode");
+  const isOpen = displayMode === "modal";
+
+  const open = (options: {
+    title?: string;
+    params?: Record<string, unknown>;
+    anchor?: { top?: number; left?: number; width?: number; height?: number };
+  }) => {
     window.openai.requestModal(options);
   };
+
+  return [open, isOpen] as const;
```

Comment:
I'm not a big fan of tupple return interfaces similar to `useState`. It was initially implemented this way in React to avoid having to rename return params all the time (as tupple destructuring is only using position to match scope variable to returned variables)

```ts
const [myCustomPropName, myCustomSetterName] = useState();
```

requires less characters than

```ts
const { value: myCustomPropName, setValue: myCustomSetterName } = useState();
```

But in our case, renaming is highly improbable, and I'd stick with more modern approaches of object being return (with the additional benefit of being able to add new props in the future on the return object without it being a breaking change - like `close` for example).


```suggestion
  return { open, isOpen }
```

Similar implementations:
- [useHooks](https://usehooks-ts.com/react-hook/use-boolean)
- [chakra](https://v2.chakra-ui.com/docs/hooks/use-disclosure)